lib/matplotlib/path.py

```diff
@@ -126,18 +126,19 @@ def __init__(self, vertices, codes=None, _interpolation_steps=1,
             and codes as read-only arrays.
         """
         vertices = _to_unmasked_float_array(vertices)
-        if vertices.ndim != 2 or vertices.shape[1] != 2:
-            raise ValueError(
-                "'vertices' must be a 2D list or array with shape Nx2")
+        cbook._check_shape((None, 2), vertices=vertices)
 
         if codes is not None:
             codes = np.asarray(codes, self.code_type)
             if codes.ndim != 1 or len(codes) != len(vertices):
                 raise ValueError("'codes' must be a 1D list or array with the "
-                                 "same length of 'vertices'")
+                                 "same length of 'vertices'. "
```

Comment:
I think the specific text here about verts and codes is worth not re-using the general checking logic.

Reply:
sounds fine